From a3571668896a600c3ffc1427201abd03d80e6b49 Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Thu, 20 Dec 2012 21:59:12 +0100
Subject: BUG/MINOR: log: add_to_logformat_list() used the wrong constants

The <type> argument was checked against LOG_FMT_* but it was passed as
LF_* which are two independant enums. It happens that the 3 first entries
in these enums do match, but this broke some experimental changes which
required another state, so let's fix this now.
---
 src/log.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/log.c b/src/log.c
index 270fd57..0d63ba6 100644
--- a/src/log.c
+++ b/src/log.c
@@ -302,7 +302,7 @@ void add_to_logformat_list(char *start, char *end, int type, struct list *list_f
 {
 	char *str;
 
-	if (type == LOG_FMT_TEXT) { /* type text */
+	if (type == LF_TEXT) { /* type text */
 		struct logformat_node *node = calloc(1, sizeof(struct logformat_node));
 		str = calloc(end - start + 1, 1);
 		strncpy(str, start, end - start);
@@ -310,7 +310,7 @@ void add_to_logformat_list(char *start, char *end, int type, struct list *list_f
 		node->arg = str;
 		node->type = LOG_FMT_TEXT; // type string
 		LIST_ADDQ(list_format, &node->list);
-	} else if (type == LOG_FMT_SEPARATOR) {
+	} else if (type == LF_SEPARATOR) {
 		struct logformat_node *node = calloc(1, sizeof(struct logformat_node));
 		node->type = LOG_FMT_SEPARATOR;
 		LIST_ADDQ(list_format, &node->list);
-- 
1.7.1

